apps/emacs/emacs.talon

```diff
@@ -0,0 +1,316 @@
+global [display] line numbers [mode]: user.emacs("global-display-line-numbers-mode")
+highlight line mode: user.emacs("hl-line-mode")
+markdown mode: user.emacs("markdown-mode")
+paredit mode: user.emacs("paredit-mode")
+rainbow mode: user.emacs("rainbow-mode")
+sub word mode: user.emacs("subword-mode")
+text mode: user.emacs("text-mode")
+visual line mode: user.emacs("visual-line-mode")
+outline minor mode: user.emacs("outline-minor-mode")
+emacs lisp mode: user.emacs("emacs-lisp-mode")
+lisp interaction mode: user.emacs("lisp-interaction-mode")
+talon script mode: user.emacs("talonscript-mode")
+
+# MACROS #
+emacs record: user.emacs("kmacro-start-macro")
+emacs stop: user.emacs("kmacro-end-macro")
+emacs play: user.emacs("kmacro-end-and-call-macro")
+
+# PROFILER #
+profiler start: user.emacs("profiler-start")
+profiler stop: user.emacs("profiler-stop")
+profiler report: user.emacs("profiler-report")
+
+# WINDOW/SPLIT MANAGEMENT #
+split solo: user.emacs("delete-other-windows")
+[split] rebalance: user.emacs("balance-windows")
+split shrink: user.emacs("shrink-window-if-larger-than-buffer")
+other [split] shrink:
+    user.split_next()
+    user.emacs("shrink-window-if-larger-than-buffer")
+    user.split_last()
+split grow: user.emacs("enlarge-window")
+split grow <number_small>:
+    user.emacs_prefix(number_small)
+    user.emacs("enlarge-window")
+split shrink <number_small>:
+    amount = number_small or 1
+    user.emacs_prefix(0 - amount)
+    user.emacs("enlarge-window")
+split widen [<number_small>]:
+    user.emacs_prefix(number_small or 1)
+    user.emacs("enlarge-window-horizontally")
+split narrow [<number_small>]:
+    user.emacs_prefix(number_small or 1)
+    user.emacs("shrink-window-horizontally")
+
+# ----- HELP ----- #
+apropos: user.emacs_help("a")
+describe (fun | function): user.emacs_help("f")
+describe key: user.emacs_help("k")
+describe key briefly: user.emacs_help("c")
+describe symbol: user.emacs_help("o")
+describe variable: user.emacs_help("v")
+describe mode: user.emacs_help("m")
+describe bindings: user.emacs_help("b")
+describe (char | character): user.emacs("describe-character")
+describe text properties: user.emacs("describe-text-properties")
+describe face: user.emacs("describe-face")
+view lossage: user.emacs_help("l")
+
+apropos <user.text>$:
+    user.emacs_help("a")
+    user.insert_formatted(text, "DASH_SEPARATED")
+    key(enter)
+describe (fun | function) <user.text>$:
+    user.emacs_help("f")
+    user.insert_formatted(text, "DASH_SEPARATED")
+    key(enter)
+describe symbol <user.text>$:
+    user.emacs_help("o")
+    user.insert_formatted(text, "DASH_SEPARATED")
+    key(enter)
+describe variable <user.text>$:
+    user.emacs_help("v")
+    user.insert_formatted(text, "DASH_SEPARATED")
+    key(enter)
+
+# ----- FILES & BUFFERS -----
+# many key bindings here are often overridden, so we don't use user.emacs,
+# eg: C-x C-f, C-x k, C-x C-b, C-x right, C-x left
+file open: key(ctrl-x ctrl-f)
```

Comment:
Should be using `user.emacs` here?

Reply:
See comment immediately above

Reply:
Oops, good point. Makes sense.

Reply:
Actually, I think it's probably ok to use `user.emacs` for these? It'll end up pressing the keybindings in emacs_command.csv, so it'll use the remapped version. I was worried about trouble with RPC, but I can use `command-remapping` to get the remapped version, *assuming* it's remapped via a command remapping rather than a straight-up keybinding... so this'll catch things like ido (overrides `switch-to-buffer`) and visual-line-mode (overrides `kill-line` to `kill-visual-line`), but not thinks like paredit, which directly overrides `ctrl-k` instead of overriding `kill-line`. This is probably fine?

Reply:
Yeah, that works for me, for consistency with the buffer commands too…